cmd/server/temporal/server.go

```diff
@@ -163,14 +162,17 @@ func (s *server) startService() common.Daemon {
 
 	if s.cfg.PublicClient.HostPort == "" {
 		log.Fatalf("need to provide an endpoint config for PublicClient")
-	} else if h, _, err := net.SplitHostPort(s.cfg.PublicClient.HostPort); err != nil || len(h) == 0 {
-		log.Fatalf("Malformed PublicClient HostPort, must be host:port - '%v' - Error - %v", s.cfg.PublicClient.HostPort, err)
 	} else {
-		connection, err := rpc.Dial(s.cfg.PublicClient.HostPort)
+		var err error
+		params.PublicClient, err = sdkclient.NewClient(sdkclient.Options{
+			HostPort:     s.cfg.PublicClient.HostPort,
+			DomainName:   common.SystemLocalDomainName,
+			MetricsScope: params.MetricScope,
+			Tracer:       opentracing.GlobalTracer(), //???
```

Comment:
@samarabbas, `opentracing.GlobalTracer()` was added in two places but all the rest did not have it. Is it harmful to always add it?

Reply:
I don't think it even matters as we don't do anything with it after moving to grpc on client side.